backend/app/main.py

```diff
@@ -24,108 +28,138 @@
-def get_listings() -> ListingsGetAllResponse:
+@app.get('/listings', 
+    response_model=ListingsGetAllResponse, responses={'500': {'model': Field500ErrorResponse}},)
+async def get_listings() -> Union[ListingsGetAllResponse, Field500ErrorResponse]:
     """
     Retrieve all listings
     """
     try:
         # Fetch all listings from MongoDB
-        listings_cursor = db.listings.find({})
-        
-        # Convert MongoDB documents to Pydantic models
-        listings = [
-            ListingsGetResponseItem(
-                id=str(listing["_id"]),  # Convert ObjectId to string
-                title=listing.get("title"),
-                price=listing.get("price"),
-                description=listing.get("description"),
-                seller_id=listing.get("user_id")
-            )
-            for listing in listings_cursor
-        ]
-
-        return listings
+        cursor = listings_collection.find()
+        listings = await cursor.to_list(length=None)  # Get all documents
+
+        response_data = []
+        for listing in listings:
+            try:
+                response_data.append(
+                    ListingsGetResponseItem(
+                        id=str(listing["_id"]),
+                        title=listing["title"],
+                        price=listing["price"],
+                        description=listing.get("description"),
+                        seller_id=listing["seller_id"],
+                        pictures=listing.get("pictures", []),
+                        condition=listing["condition"],
+                        category=listing.get("category"),
+                        date_posted=listing.get("date_posted"),  # need to decide on the date format
+                        campus=listing.get("campus"),
+                    )
+                )
+            except Exception as e:
+                pass
+            #   print(f"Skipping invalid listing {listing['_id']}: {e}")  # i kept this for later and we can potentially use it for logging 
+
+        return ListingsGetAllResponse(listings=response_data, total=len(response_data))
 
     except Exception as e:
-        raise HTTPException(status_code=500, detail=f"Unexpected error: {str(e)}")
+        return Field500ErrorResponse(error="Internal Server Error. Please try again later.")
 
 
 @app.post(
-    '/listings', response_model=None, responses={'201': {'model': ListingsPostResponse}}
+    '/listings',
+    response_model=None,
+    responses={
+        '201': {'model': ListingsPostResponse},
+        '500': {'model': Field500ErrorResponse},
+    },
 )
-def post_listings(body: ListingsPostRequest) -> Optional[ListingsPostResponse]:
+async def post_listings(
+    body: ListingsPostRequest,
+) -> Optional[Union[ListingsPostResponse, Field500ErrorResponse]]:
     """
     Create a new listing
     """
     try:
-        # Prepare listing document
+        # Prepare data for MongoDB
         listing_data = body.dict()
```

Comment:
One thing we should try checking is if `required` fields are automatically validated for. Either way, there must be some kind of error handling in case a supposedly `required` field is not provided.